tests/python_frontend/context_manager_test.py

```diff
@@ -36,5 +36,56 @@ def prog(A: dace.float64[20]):
     assert ctx.should_pass
 
 
+def test_ctxmgr_name_clash():
+    
+    from context_managers.context_a import my_dace_ctxmgr_program as prog_a
+    from context_managers.context_b import my_dace_ctxmgr_program as prog_b
+
+    rng = np.random.default_rng(42)
+
+    def dace_blocker(f):
+        return f
+    
+    @dace_blocker
+    def randint():
+        return rng.integers(0, 2)
+
+    @dace.program
+    def ctxmgr_name_clashing_0():
+        i: dace.int64 = randint()
+        if i == 0:
+            prog_a()
+        else:
+            prog_b()
+        return i
+
+    @dace.program
+    def ctxmgr_name_clashing_1():
+        i: dace.int64 = randint()
+        if i == 0:
+            prog_a()
+        else:
+            prog_b()
+        return i
+    
+    sdfg = ctxmgr_name_clashing_0.to_sdfg()
+    
+    for i, f in enumerate([ctxmgr_name_clashing_0, ctxmgr_name_clashing_1]):
+
+        if i > 0:
+            f.load_precompiled_sdfg(sdfg.build_folder)
+
+        a_count = 0
+        b_count = 0
+        for _ in range(100):
```

Comment:
This test looks overly complicated. Any reason?

Reply:
Yes and no. I was testing various modifications but the test combines all the reported issues:
- context managers with potentially the same name (when auto simplification is on)
- reloading from file and calling through another program 

If you are asking about the `count`s, it is just to confirm that both context managers are executed despite having the same name. It could be simplified of course.